### PR TITLE
tileset parser fix for ObjectGroup

### PR DESCRIPTION
**Pull request by lfraker**
Fix to support parsing ObjectGroups from TileSets, bug raised in issue: #31
Included test coverage for ObjectGroup on TileSet case.

### CODE REVIEW

**Review by spydon — APPROVED** [review]
Lgtm, thanks for the contribution!

**Comment by spydon** [comment]
If you bump the sdk from 2.12 to 2.15 in the pubspec file I think that it will solve the failing checks.

File: lib/src/tileset/tile.dart [hunk 1]
```diff
@@ -56,7 +56,7 @@ class Tile {
               .toList() ??
           [],
       image: parser.getSingleChildOrNullAs('image', TiledImage.parse),
-      objectGroup: parser.getSingleChildOrNullAs('objectGroup', Layer.parse),
+      objectGroup: parser.getSingleChildOrNullAs('objectgroup', Layer.parse),
```
Comment (luanpotter) [inline lib/src/tileset/tile.dart:59]: note: important to make sure the this works with json as well.
looking at [the docs[(https://doc.mapeditor.org/en/stable/reference/json-map-format/) it does appear so, but we dont have any json example with that field. would be easy to export you fixture map as json and add a test for that too?

**Review by luanpotter — APPROVED** [review]
LGTM

File: lib/src/tileset/tile.dart (hunk 1 above)
Comment (lfraker) [reply lib/src/tileset/tile.dart:59]: @luanpotter, I didn't get a chance to update this today, but I can add a test to include json before completing the PR tomorrow. It looks like it should work, and not too much work to include another test.

Comment (lfraker) [reply lib/src/tileset/tile.dart:59]: @spydon did you want me to update the version number still, or just complete the PR with check failures, and you'll merge into your PR?

Comment (lfraker) [reply lib/src/tileset/tile.dart:59]: @spydon did you want me to update the version number still. Or did you want me to leave the PR with check failures, and you'll merge into your PR with the updated version?

Comment (spydon) [reply lib/src/tileset/tile.dart:59]: I updated this branch to use the new cicd structure and now everything passes. We'd be really happy if you did a PR for the other issues that you found too! :)

**Pull request merged**